librz/util/regex.c

```diff
@@ -0,0 +1,250 @@
+// SPDX-FileCopyrightText: 2023 Rot127 <unisono@quyllur.org>
+// SPDX-License-Identifier: LGPL-3.0-only
+
+#include <rz_vector.h>
+#include <rz_regex.h>
+#include <rz_types.h>
+#include <rz_util/rz_assert.h>
+#include <rz_util.h>
+
+/**
+ * \file Defines the wrapper functions to PCRE2.
+ */
+
+/**
+ * \brief Compile a regex pattern to a RzRegex and return it.
+ * In case of an error, an error message is printed and NULL is returned.
+ *
+ * \param pattern The regex pattern string.
+ * \param cflags The compilation flags or zero for default.
+ *
+ * \return The compiled regex or NULL in case of failure.
+ */
+RZ_API RZ_OWN RzRegex *rz_regex_new(const char *pattern, RzRegexFlags cflags) {
+	RzRegexStatus err_num;
+	RzRegexSize err_off;
+	RzRegex *regex = pcre2_compile(
+		(PCRE2_SPTR)pattern,
+		PCRE2_ZERO_TERMINATED,
+		cflags | PCRE2_UTF,
+		&err_num,
+		&err_off,
+		NULL);
+	if (!regex) {
+		PCRE2_UCHAR buffer[256];
+		pcre2_get_error_message(err_num, buffer, sizeof(buffer));
+		RZ_LOG_ERROR("Regex compilation failed at offset %" PFMTSZu ": %s\n", err_off,
```

Comment:
```suggestion
		RZ_LOG_ERROR("Regex compilation failed at %" PFMTSZu ": %s\n", err_off,
```
is a char, not an "offset" as we might refer addresses in rizin.